gettsim/benefits/elterngeld.py

```diff
@@ -0,0 +1,215 @@
+
+    in_cols = list(household.columns.values)
+    # Everything was already initialized
+    out_cols = []
+    household.loc[household["elternzeit_anspruch"], :] = _apply_tax_transfer_func(
+        household[household["elternzeit_anspruch"]],
+        tax_func=calc_elterngeld,
+        level=["hid", "tu_id", "pid"],
+        in_cols=in_cols,
+        out_cols=out_cols,
+        func_kwargs={
+            "params": params,
+            "soz_vers_beitr_params": soz_vers_beitr_params,
+            "e_st_abzuege_params": e_st_abzuege_params,
+            "e_st_params": e_st_params,
+            "soli_st_params": soli_st_params,
+        },
+    )
+
+    household.loc[~household["elternzeit_anspruch"], ["elterngeld"]] = 0
+
+    return household
+
+
+def calc_elterngeld(
+    person,
+    params,
+    soz_vers_beitr_params,
+    e_st_abzuege_params,
+    e_st_params,
+    soli_st_params,
+):
+    """ Calculating elterngeld given the relevant wage and the eligibility on sibling
+    bonus.
+
+    """
+    considered_wage = calc_considered_wage(
+        person,
+        params,
+        soz_vers_beitr_params,
+        e_st_abzuege_params,
+        e_st_params,
+        soli_st_params,
+    )
+    if considered_wage < 0:
+        person["elterngeld"] = 0
+    else:
+        payed_percentage = calc_elterngeld_percentage(considered_wage, params)
+
+        elterngeld_calc = considered_wage * payed_percentage
+
+        prelim_elterngeld = max(
+            min(elterngeld_calc, params["elgmax"]), params["elgmin"]
+        )
+        if person["geschw_bonus"]:
+            prelim_elterngeld += calc_geschw_bonus(elterngeld_calc, params)
+        prelim_elterngeld += person["num_mehrlinge"] * params["mehrling_bonus"]
+        person["elterngeld"] = prelim_elterngeld
+
+    return person
+
+
+def calc_considered_wage(
+    person,
+    params,
+    soz_vers_beitr_params,
+    e_st_abzuege_params,
+    e_st_params,
+    soli_st_params,
+):
+    """ Calculating the relevant wage for the calculation of elterngeld.
+
+
+    According to § 2 (1) BEEG elterngeld is calculated by the loss of income due to
+    child raising.
+    """
+    # Beitragsbemessungsgrenze differs in east and west germany
+    westost = "o" if person["east"] else "w"
+
+    net_wage_last_year = proxy_net_wage_last_year(
+        person,
+        e_st_params,
+        soli_st_params,
+        beit_bem_grenz=soz_vers_beitr_params[f"rvmaxek{westost}"],
+        werbungs_pausch=e_st_abzuege_params["werbung"],
+        soz_vers_pausch=params["soz_vers_pausch"],
+    )
+
+    current_net_wage = calc_net_wage(person)
+
+    return net_wage_last_year - current_net_wage
+
+
+def calc_elterngeld_percentage(considered_wage, params):
+    """ This function calculates the percentage share of net income, which is
+    reinbursed in the time of recieving elterngeld.
+
+    According to § 2 (2) BEEG the percentage increases below elg_st_1 and decreases
+    above elg_st_2.
```

Comment:
"According to ..., the percentage increases for low incomes and decreases for high incomes until its minimum of 65%.

Reply:
Changed comment, but didn't include specific values and instead the variable names.